### ISSUE TITLE
SEGV (11) on SearchPanel for wxhexeditor-0.23+git20170214-24.1

### ISSUE THREAD

**Issue by MikhailKasimov** [EUA/wxHexEditor#38]
Hello!

've caught another SEGV (11), when tried to switch between items in search-result panel.

```
k_mikhail@linux-mk500:~> coredumpctl | grep wxH
Суб 2017-02-18 07:57:44 EET   3855  1000   100  11 * /usr/bin/wxHexEditor
```
```
k_mikhail@linux-mk500:~> coredumpctl gdb 3855
           PID: 3855 (wxHexEditor)
           UID: 1000 (k_mikhail)
           GID: 100 (users)
        Signal: 11 (SEGV)
     Timestamp: Суб 2017-02-18 07:57:21 EET (10min ago)
  Command Line: wxHexEditor
    Executable: /usr/bin/wxHexEditor
 Control Group: /user.slice/user-1000.slice/session-2.scope
          Unit: session-2.scope
         Slice: user-1000.slice
       Session: 2
     Owner UID: 1000 (k_mikhail)
       Boot ID: 1ce14bb138ad4e279a2e2d33226baf0c
    Machine ID: 4664bc7cba6aee90fe1efc7f571cbf28
      Hostname: linux-mk500
      Coredump: /var/lib/systemd/coredump/core.wxHexEditor.1000.1ce14bb138ad4e279a2e2d33226baf0c.3855.1487397441000000.xz
       Message: Process 3855 (wxHexEditor) of user 1000 dumped core.

GNU gdb (GDB; openSUSE Leap 42.2) 7.11.1
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-suse-linux".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://bugs.opensuse.org/>.
Find the GDB manual and other documentation resources online at:
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from /usr/bin/wxHexEditor...(no debugging symbols found)...done.
[New LWP 3855]
[New LWP 3931]
[New LWP 3972]
[New LWP 3932]
[New LWP 3933]
[New LWP 3974]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib64/libthread_db.so.1".
Core was generated by `wxHexEditor'.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00000000004688a0 in SearchPanel::OnTagSelect(wxCommandEvent&) ()
[Current thread is 1 (Thread 0x7f09b0724a80 (LWP 3855))]
(gdb)
```
BackTrace:
```
(gdb) bt
#0  0x00000000004688a0 in SearchPanel::OnTagSelect(wxCommandEvent&) ()
#1  0x00007f09aed4b95e in wxAppConsoleBase::CallEventHandler(wxEvtHandler*, wxEventFunctor&, wxEvent&) const ()
    at /usr/lib64/libwx_baseu-suse.so.2
#2  0x00007f09aee8f872 in wxEvtHandler::ProcessEventIfMatchesId(wxEventTableEntryBase const&, wxEvtHandler*, wxEvent&) ()
    at /usr/lib64/libwx_baseu-suse.so.2
#3  0x00007f09aee904b9 in wxEvtHandler::SearchDynamicEventTable(wxEvent&) () at /usr/lib64/libwx_baseu-suse.so.2
#4  0x00007f09aee9050e in wxEvtHandler::TryHereOnly(wxEvent&) () at /usr/lib64/libwx_baseu-suse.so.2
#5  0x00007f09aee905a3 in wxEvtHandler::ProcessEventLocally(wxEvent&) () at /usr/lib64/libwx_baseu-suse.so.2
#6  0x00007f09aee905f9 in wxEvtHandler::ProcessEvent(wxEvent&) () at /usr/lib64/libwx_baseu-suse.so.2
#7  0x00007f09aee9068c in wxEvtHandler::SafelyProcessEvent(wxEvent&) () at /usr/lib64/libwx_baseu-suse.so.2
#8  0x00007f09af9229b7 in wxListBoxBase::SendEvent(int, int, bool) () at /usr/lib64/libwx_gtk2u_core-suse.so.2
#9  0x00007f09aca49b37 in _g_closure_invoke_va (closure=closure@entry=0x22234b0, return_value=return_value@entry=0x0, instance=instance@entry=0x2222440, args=args@entry=0x7ffe1f00f148, n_params=0, param_types=0x0) at gclosure.c:867
#10 0x00007f09aca62918 in g_signal_emit_valist (instance=0x2222440, signal_id=<optimized out>, detail=0, var_args=var_args@entry=0x7ffe1f00f148) at gsignal.c:3294
#11 0x00007f09aca63592 in g_signal_emit (instance=<optimized out>, signal_id=<optimized out>, detail=detail@entry=0)
    at gsignal.c:3441
#12 0x00007f09ad5c27bc in _gtk_tree_selection_internal_select_node (selection=<optimized out>, node=<optimized out>, tree=<optimized out>, path=path@entry=0x237a250, mode=<optimized out>, override_browse_mode=override_browse_mode@entry=0)
    at gtktreeselection.c:1427
#13 0x00007f09ad5d64ae in gtk_tree_view_real_set_cursor (tree_view=tree_view@entry=0x21d4740 [GtkTreeView], path=path@entry=0x237a250, clear_and_select=clear_and_select@entry=1, clamp_node=1) at gtktreeview.c:12610
#14 0x00007f09ad5daa38 in gtk_tree_view_button_press (widget=0x21d4740 [GtkTreeView], event=0x24815c0) at gtktreeview.c:2797
#19 0x00007f09aca63592 in <emit signal ??? on instance 0x21d4740 [GtkTreeView]> (instance=instance@entry=0x21d4740, signal_id=<optimized out>, detail=detail@entry=0) at gsignal.c:3441
    #15 0x00007f09ad4e2b65 in _gtk_marshal_BOOLEAN__BOXED (closure=0x1e23ff0, return_value=0x7ffe1f00f480, n_param_values=<optimized out>, param_values=0x7ffe1f00f530, invocation_hint=<optimized out>, marshal_data=0x7f09ad5da240 <gtk_tree_view_button_press>) at gtkmarshalers.c:86
    #16 0x00007f09aca49908 in g_closure_invoke (closure=closure@entry=0x1e23ff0, return_value=return_value@entry=0x7ffe1f00f480, n_param_values=2, param_values=param_values@entry=0x7ffe1f00f530, invocation_hint=invocation_hint@entry=0x7ffe1f00f4d0)
    at gclosure.c:804
    #17 0x00007f09aca5b0cb in signal_emit_unlocked_R (node=node@entry=0x1e4dbb0, detail=detail@entry=0, instance=instance@entry=0x21d4740, emission_return=emission_return@entry=0x7ffe1f00f600, instance_and_params=instance_and_params@entry=0x7ffe1f00f530) at gsignal.c:3667
    #18 0x00007f09aca62f78 in g_signal_emit_valist (instance=<optimized out>, signal_id=<optimized out>, detail=<optimized out>, var_args=var_args@entry=0x7ffe1f00f6c8) at gsignal.c:3395
#20 0x00007f09ad5f2884 in gtk_widget_event_internal (widget=widget@entry=0x21d4740 [GtkTreeView], event=event@entry=0x24815c0) at gtkwidget.c:5017
#21 0x00007f09ad5f2b59 in IA__gtk_widget_event (widget=widget@entry=0x21d4740 [GtkTreeView], event=event@entry=0x24815c0)
    at gtkwidget.c:4814
#22 0x00007f09ad4e1314 in IA__gtk_propagate_event (widget=0x21d4740 [GtkTreeView], event=0x24815c0) at gtkmain.c:2501
#23 0x00007f09ad4e16cb in IA__gtk_main_do_event (event=0x24815c0) at gtkmain.c:1696
#24 0x00007f09ad153c1c in gdk_event_dispatch (source=<optimized out>, callback=<optimized out>, user_data=<optimized out>)
    at gdkevents-x11.c:2425
#25 0x00007f09ac775134 in g_main_context_dispatch (context=0x1e09470) at gmain.c:3154
#26 0x00007f09ac775134 in g_main_context_dispatch (context=context@entry=0x1e09470) at gmain.c:3769
#27 0x00007f09ac775388 in g_main_context_iterate (context=0x1e09470, block=block@entry=1, dispatch=dispatch@entry=1, self=<optimized out>) at gmain.c:3840
#28 0x00007f09ac77564a in g_main_loop_run (loop=0x23b98c0) at gmain.c:4034
#29 0x00007f09ad4e07b7 in IA__gtk_main () at gtkmain.c:1268
#30 0x00007f09af809c95 in wxGUIEventLoop::DoRun() () at /usr/lib64/libwx_gtk2u_core-suse.so.2
#31 0x00007f09aed7cad1 in wxEventLoopBase::Run() () at /usr/lib64/libwx_baseu-suse.so.2
#32 0x00007f09aed4bf0d in wxAppConsoleBase::MainLoop() () at /usr/lib64/libwx_baseu-suse.so.2
#33 0x00007f09aedc642d in wxEntry(int&, wchar_t**) () at /usr/lib64/libwx_baseu-suse.so.2
#34 0x000000000041bff2 in main ()
(gdb)
```
libwx_ 3.1.0-4.1 version is installed for wxhexeditor-0.23+git20170214-24.1.

**Comment by MikhailKasimov** [comment]
Seems, I've got the nature of this bug:

1) Open some file, for example, some MZ-PE.
2) Put 4D5A and push 'Find All'. Result: you'll get some info in 'Search Results' panel.
3) Put some query, which is guaranteed missed in this file - e.g. 7F454C46 and push 'Find All' again. Result: you'll get 0 successful ("No matches found"), but 'Search Results' panel will contain results of your previous query (for 4D5A). And when you try to navigate on this panel, you'll get segv (11).

**Comment by plater** [comment]
This is found in revision#ec653348e2f59f374a0a0e71e4ea6d97faa6a990

**Comment by EUA** [comment]
Good catch @MikhailKasimov ,
Fixed in fa8e5a2
Thanks
